Log timeout in ProcessGroupNCCL for NCCL_BLOCKING_WAIT

rohan-varma opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#47895 Log timeout in ProcessGroupNCCL for NCCL_BLOCKING_WAIT**

It would be useful in error messages to have the timeout in addition
to the operation timed out message.

Differential Revision: [D24943318](https://our.internmc.facebook.com/intern/diff/D24943318/)

rohan-varma committed:
Log timeout in ProcessGroupNCCL for NCCL_BLOCKING_WAIT

It would be useful in error messages to have the timeout in addition
to the operation timed out message.

Differential Revision: [D24943318](https://our.internmc.facebook.com/intern/diff/D24943318/)

[ghstack-poisoned]

osalpekar commented:
We're currently landing https://github.com/pytorch/pytorch/pull/47246 which logs the timeout and some other useful metadata/error information. I'm guessing we should be able to close this PR since this functionality is being covered?

rohan-varma committed:
Update on "Log timeout in ProcessGroupNCCL for NCCL_BLOCKING_WAIT"


It would be useful in error messages to have the timeout in addition
to the operation timed out message.

Differential Revision: [D24943318](https://our.internmc.facebook.com/intern/diff/D24943318/)

[ghstack-poisoned]

rohan-varma commented:
@osalpekar Sounds good, looks like that PR covers this.
